tinylog-impl/src/main/java/org/tinylog/writers/JsonWriter.java

```diff
@@ -0,0 +1,268 @@
+	private FileChannel fileChannel;
+	private FileChannel inputChannel;
+
+	private StringBuilder builder;
+	private final Token messageToken;
+	private final Token timestampToken;
+	private final Token methodToken;
+	private final Token levelToken;
+	private final Token classToken;
+	private final Token threadToken;
+
+	private byte[] commaByte;
+	private byte[] bracketOpenByte;
+	private byte[] bracketCloseByte;
+
+	/**
+	 * @throws IOException              File not found or couldn't access file
+	 * @throws IllegalArgumentException Log file is not defined in configuration
+	 */
+	public JsonWriter() throws IOException {
+		this(Collections.<String, String>emptyMap());
+	}
+
+	/**
+	 * @param properties Configuration for writer
+	 * @throws IOException              File not found or couldn't access file
+	 * @throws IllegalArgumentException Log file is not defined in configuration
+	 */
+	public JsonWriter(final Map<String, String> properties) throws IOException {
+
+		String exceptionFilter = properties.get("exception");
+		messageToken = new FormatPatternParser(exceptionFilter).parse(MESSAGE_PATTERN);
+		timestampToken = new FormatPatternParser(exceptionFilter).parse(TIMESTAMP_PATTERN);
+		methodToken = new FormatPatternParser(exceptionFilter).parse(METHOD_PATTERN);
+		levelToken = new FormatPatternParser(exceptionFilter).parse(LEVEL_PATTERN);
+		classToken = new FormatPatternParser(exceptionFilter).parse(CLASS_PATTERN);
+		threadToken = new FormatPatternParser(exceptionFilter).parse(THREAD_PATTERN);
+
+		charset = getCharset(properties);
+		commaByte = ",".getBytes(charset);
+		bracketOpenByte = "[".getBytes(charset);
+		bracketCloseByte = "]".getBytes(charset);
+
+		String fileName = getFileName(properties);
+		File file = new File(fileName).getAbsoluteFile();
+		file.getParentFile().mkdirs();
+
+		boolean append = Boolean.parseBoolean(properties.get("append"));
+		FileOutputStream stream = new FileOutputStream(file, append);
+
+		fileChannel = stream.getChannel();
+
+		writer = new OutputStreamWriter(stream);
+
+		boolean buffered = Boolean.parseBoolean(properties.get("buffered"));
+		if (buffered) {
+			writer = new BufferedWriterDecorator(writer);
+		}
+
+		boolean writingThread = Boolean.parseBoolean(properties.get("writingthread"));
+		if (!writingThread) {
+			writer = new SynchronizedWriterDecorator(writer, stream);
+		}
+		FileInputStream inputStream = null;
+		try {
+			inputStream = new FileInputStream(file);
+			inputChannel = inputStream.getChannel();
+			preprocessFile(append);
+		} finally {
+			if (inputStream != null) {
+				inputStream.close();
+			}
+		}
+	}
+
+	private Charset getCharset(final Map<String, String> properties) {
+		String charsetName = properties.get("charset");
+		try {
+			return charsetName == null ? Charset.defaultCharset() : Charset.forName(charsetName);
+		} catch (IllegalArgumentException ex) {
+			InternalLogger.log(Level.ERROR, "Invalid charset: " + charsetName);
+			return Charset.defaultCharset();
```

Comment:
Could you add a JUnit test for the error case?

Reply:
Fixed in https://github.com/tinylog-org/tinylog/pull/195/commits/a849524287ae1533012edf7851b32d03b7e289ef